examples~/quickstart/client/Program.cs

```diff
@@ -88,44 +96,59 @@ void PrintMessage(Message message)
     Console.WriteLine($"{senderName}: {message.Text}");
 }
 
-void Message_OnInsert(Message insertedValue, ReducerEvent? dbEvent)
+void Message_OnInsert(EventContext ctx, Message insertedValue)
 {
-    if (dbEvent != null)
+    if (ctx.Reducer is not Event<Reducer>.SubscribeApplied)
     {
         PrintMessage(insertedValue);
     }
 }
 
-void Reducer_OnSetNameEvent(ReducerEvent reducerEvent, string name)
+void Reducer_OnSetNameEvent(EventContext ctx, string name)
 {
-    if (reducerEvent.Identity == local_identity && reducerEvent.Status is UpdateStatus.Failed)
+    if (ctx.Reducer is Event<Reducer>.Reducer reducer)
     {
-        Console.Write($"Failed to change name to {name}");
+        var e = reducer.ReducerEvent;
+        if (e.CallerIdentity == local_identity && e.Status is Status.Failed(var error))
+        {
+            Console.Write($"Failed to change name to {name}: {error}");
+        }
     }
 }
 
-void Reducer_OnSendMessageEvent(ReducerEvent reducerEvent, string text)
+void Reducer_OnSendMessageEvent(EventContext ctx, string text)
 {
-    if (reducerEvent.Identity == local_identity && reducerEvent.Status is UpdateStatus.Failed)
+    if (ctx.Reducer is Event<Reducer>.Reducer reducer)
     {
-        Console.Write($"Failed to send message {text}");
+        var e = reducer.ReducerEvent;
+        if (e.CallerIdentity == local_identity && e.Status is Status.Failed(var error))
+        {
+            Console.Write($"Failed to send message {text}: {error}");
+        }
     }
 }
 
-void OnConnect()
+void OnConnect(Identity identity, string authToken)
```

Comment:
This and OnDisconnect signatures don't match the proposal - user needs access to the DbConnection.

Reply:
Are we looking at the same version of the proposal? The one I have only has these two arguments.